packages/core/src/middleware/koa-spa-proxy.ts

```diff
@@ -77,15 +80,17 @@ export default function koaSpaProxy<StateT, ContextT extends IRouterParamContext
     }
 
     if (!EnvSet.values.isProduction) {
-      return spaProxy(ctx, next);
+      return devProxy(ctx, next);
     }
 
+    const distributionPath = await getDistributionPath(packagePath, ctx);
+
     const spaDistributionFiles = await fs.readdir(distributionPath);
 
     if (!spaDistributionFiles.some((file) => requestPath.startsWith('/' + file))) {
       ctx.request.path = '/';
     }
 
-    return spaProxy(ctx, next);
+    return serveStatic(distributionPath)(ctx, next);
```

Comment:
we may add a header like `Logto-Experience-Version` to specify which version it's using

Reply:
Updated added a Logto-Experience-Package header in the `getDistributionPath` method.

Reply:
<img width="507" alt="image" src="https://github.com/user-attachments/assets/618e0e52-1a8e-44d6-aa6c-85d3e88ff425">